File: code_examples/3-Structured_Problem_Solving/10-Try_Catch/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module checks for a runtime error
"""


def main() -> None:
    """The main() function checks for a runtime error, returns None."""

    # input
    integer_as_string = input("Enter an integer: ")

    # process & output
    try:
        integer_as_number = int(integer_as_string)
    except ValueError:
        print(f"The error was {ValueError}.")
    else:
        print(f"You entered: {integer_as_number}.")
    finally:
        print("Thanks for playing.")

    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/3-Structured_Problem_Solving/10-Try_Catch/Java/Main.java
/*
 * This program uses a try statement
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    // this function uses a try statement

    // create Scanner object for user input
    Scanner scanner = new Scanner(System.in);

    // input
    System.out.print("Enter an integer: ");
    String integerAsString = scanner.nextLine();

    // process & output
    try {
        int integerAsNumber = Integer.parseInt(integerAsString);
        System.out.printf("The integer was: %d.%n", integerAsNumber);
    } catch (NumberFormatException e) {
        System.out.printf("The error was: %s.%n", e.getMessage());
    } finally {
        System.out.println("Thanks for playing.");
    }

    // close the Scanner object
    scanner.close();
    System.out.println("\nDone.");
  }
}
